indev/libinput.c

```diff
@@ -410,39 +491,51 @@ static void reset_scanned_devices(void) {
 static void read_pointer(libinput_drv_state_t *state, struct libinput_event *event) {
   struct libinput_event_touch *touch_event = NULL;
   struct libinput_event_pointer *pointer_event = NULL;
+  libinput_lv_event_t *evt = NULL;
   enum libinput_event_type type = libinput_event_get_type(event);
 
   /* We need to read unrotated display dimensions directly from the driver because libinput won't account
    * for any rotation inside of LVGL */
   lv_disp_drv_t *drv = lv_disp_get_default()->driver;
 
   switch (type) {
+    // We get a touch down and then motion event, we want to squash them
     case LIBINPUT_EVENT_TOUCH_MOTION:
+      evt = peek_event(state);
+      __attribute__((fallthrough));
     case LIBINPUT_EVENT_TOUCH_DOWN:
+      if (!evt)
+        evt = new_event(state);
       touch_event = libinput_event_get_touch_event(event);
+      //evt = &state->points[libinput_event_touch_get_seat_slot(touch_event)];
       lv_coord_t x = libinput_event_touch_get_x_transformed(touch_event, drv->physical_hor_res > 0 ? drv->physical_hor_res : drv->hor_res) - drv->offset_x;
       lv_coord_t y = libinput_event_touch_get_y_transformed(touch_event, drv->physical_ver_res > 0 ? drv->physical_ver_res : drv->ver_res) - drv->offset_y;
       if (x < 0 || x > drv->hor_res || y < 0 || y > drv->ver_res) {
         break; /* ignore touches that are out of bounds */
       }
-      state->most_recent_touch_point.x = x;
-      state->most_recent_touch_point.y = y;
-      state->button = LV_INDEV_STATE_PR;
+      evt->point.x = x;
+      evt->point.y = y;
+      evt->pressed = LV_INDEV_STATE_PR;
       break;
     case LIBINPUT_EVENT_TOUCH_UP:
-      state->button = LV_INDEV_STATE_REL;
+      evt = new_event(state);
+      touch_event = libinput_event_get_touch_event(event);
```

Comment:
I think this isn't actually used anymore because the line below is commented out?